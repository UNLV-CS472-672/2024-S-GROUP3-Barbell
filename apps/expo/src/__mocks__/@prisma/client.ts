export const ChatType = {
  DIRECT: 'DIRECT',
  GROUP: 'GROUP',
}
