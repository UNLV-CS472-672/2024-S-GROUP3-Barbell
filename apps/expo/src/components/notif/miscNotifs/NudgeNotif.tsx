import { View, Text, TouchableOpacity } from "react-native"
import { MaterialCommunityIcons } from '@expo/vector-icons'
import TimeAgo from "~/components/timeAgo/TimeAgo"
// TODO: change from any type to notification type once schema completed
//////////////////////////////////////////////////////////////////////////
// can delete from here ...
import userData from "@/packages/db/src/mock-data/user.json"
const userInfo: any[] = userData
function getUsernameFromSenderId(senderId: number): string {
  const user = userInfo.find((item) => item.id === senderId);
  if (user) {
    return user.username;
  } else {
    return String(senderId);
  }
}
// to here once we get the API setup to get username from sender ID
//////////////////////////////////////////////////////////////////////////

export default function NudgeNotif({ notif }: { notif: any }) {

  return (
    <View>
      <View className="flex flex-row items-center ml-3 mr-3 mt-4">
        {/*flexing arm is the placeholder, might change in future*/}
        <View className="flex flex-col">
          <View className="mr-2">
            <MaterialCommunityIcons name="arm-flex" size={56} color="#CACACA" />
          </View>
        </View>
        
        <View className="flex flex-col flex-1">
          {/*nudged text*/}
          <Text className="mr-2 mb-2" style={{color: "#CACACA"}}>{getUsernameFromSenderId(notif.senderId)} {notif.content}</Text>
          {/*nudge back button*/}
          <TouchableOpacity className="py-2 px-4 rounded-lg flex-1" style={{backgroundColor: "#CACACA"}}>
            <Text style={{color: "#1C1B1B"}} className="text-center">Nudge back</Text>
          </TouchableOpacity>
          <TimeAgo notif={notif} />
        </View>
      </View>
      {/*thin line between notifications*/}
      <View className="ml-3 mr-3" style={{borderBottomWidth: 1, borderBottomColor: "#737272"}}/>
    </View>
  )
}