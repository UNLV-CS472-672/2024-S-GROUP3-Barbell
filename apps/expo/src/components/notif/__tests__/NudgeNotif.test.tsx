import React from 'react'
import { fireEvent, render, screen } from '@testing-library/react-native'
import NudgeNotif from '~/components/notif/miscNotifs/NudgeNotif'
import { Notification } from '@prisma/client'

test('NudgeNotif', async () => {
  const expectedNotif: Notification = {
    "id": 3,
    "createdAt": new Date("2024-03-14T10:15:00Z"),
    "content": "nudged you",
    "type": "NUDGE",
    "read": false,
    "receiverId": 4,
    "senderId": 2
  }
  const expectedSenderUsername = 'username'
  const expectedContent = expectedNotif.content

  render(<NudgeNotif notif={expectedNotif} senderUsername={expectedSenderUsername} />)
  const senderUsernameOutput = await screen.findByTestId('sender-username-with-content')

  expect(senderUsernameOutput).toHaveTextContent(expectedSenderUsername + ' ' + expectedContent)
  expect(screen.toJSON()).toMatchSnapshot()
})
