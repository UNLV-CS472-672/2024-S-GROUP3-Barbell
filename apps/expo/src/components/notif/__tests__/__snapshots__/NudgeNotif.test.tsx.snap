// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NudgeNotif 1`] = `
<View
  testID="nudge-notif-container"
>
  <View
    className="ml-3 mr-3 mt-4 flex flex-row items-center"
  >
    <View
      className="flex flex-col"
    >
      <View
        className="mr-2"
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "#CACACA",
                "fontSize": 56,
              },
              undefined,
              {
                "fontFamily": "material-community",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          ó°¿—
        </Text>
      </View>
    </View>
    <View
      className="flex flex-1 flex-col"
    >
      <Text
        className="mb-2 mr-2"
        style={
          {
            "color": "#CACACA",
          }
        }
        testID="sender-username-with-content"
      >
        username
         
        nudged you
      </Text>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#CACACA",
            "opacity": 1,
          }
        }
      >
        <Text
          className="text-center"
          style={
            {
              "color": "#1C1B1B",
            }
          }
        >
          Nudge back
        </Text>
      </View>
      <Text
        className="pb-2 pt-2 text-right"
        style={
          {
            "color": "#CACACA",
            "fontSize": 11,
          }
        }
        testID="timeAgoText"
      >
        3 weeks ago
      </Text>
    </View>
  </View>
  <View
    className="ml-3 mr-3"
    style={
      {
        "borderBottomColor": "#737272",
        "borderBottomWidth": 1,
      }
    }
  />
</View>
`;
